exporter/newrelicexporter/factory.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package newrelicexporter
+
+import (
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"go.uber.org/zap"
+)
+
+const typeStr = "newrelic"
+
+// Factory is the factory for the New Relic exporter.
+type Factory struct{}
+
+// Type gets the type of the exporter configuration created by this factory.
+func (f *Factory) Type() configmodels.Type {
+	return configmodels.Type(typeStr)
+}
+
+// CreateDefaultConfig creates a default configuration for this exporter.
+func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
+	return &Config{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: configmodels.Type(typeStr),
+			NameVal: typeStr,
+		},
+		Timeout: time.Second * 15,
+	}
+}
+
+// CreateTraceExporter creates a New Relic trace exporter for this configuration.
+func (f *Factory) CreateTraceExporter(logger *zap.Logger, cfg configmodels.Exporter) (component.TraceExporterOld, error) {
```

Comment:
Can you please use the new components? We are planning to change everything to that, so no point in merging old components anymore

Reply:
Okay, will do.

Reply:
@bogdandrutu can we keep using the old components until https://github.com/open-telemetry/opentelemetry-collector/issues/986 is resolved or has been discussed more?

Reply:
@bogdandrutu: based on https://github.com/open-telemetry/opentelemetry-collector/issues/1039#issuecomment-635402224, can you clarify what you are asking for here? It sounds like the new components are still a work in progress, and I'm realizing I may not understand what the ask was here.

Reply:
Talking with @bogdandrutu in Gitter, the conclusion was to keep using the "old" stuff and prioritize switching when the OTLP and then the new pdata package are ready to go.